lib/api/route/channels.dart

```diff
@@ -38,3 +39,30 @@ class GetStreamTopicsEntry {
 
   Map<String, dynamic> toJson() => _$GetStreamTopicsEntryToJson(this);
 }
+
+/// https://zulip.com/api/update-user-topic
+/// https://zulip.com/api/mute-topic (legacy)
+Future<void> updateUserTopic(ApiConnection connection, {
+  required int streamId,
+  required String topic,
+  required UserTopicVisibilityPolicy visibilityPolicy,
+}) {
+  assert(visibilityPolicy != UserTopicVisibilityPolicy.unknown);
+  if (connection.zulipFeatureLevel! < 170) {
+    // TODO(server-7): deprecated in FL 170
+    assert(visibilityPolicy == UserTopicVisibilityPolicy.none
+           || visibilityPolicy == UserTopicVisibilityPolicy.muted);
+    final op = visibilityPolicy == UserTopicVisibilityPolicy.none ? 'remove'
+                                                                  : 'add';
+    return connection.patch('muteTopic', (_) {}, 'users/me/subscriptions/muted_topics', {
```

Comment:
Yeah. The story is that the caller of this is not expected to handle the error, mostly because it isn't helpful to the user. If the topic is already muted/unmuted, and that the user wants it to be muted/unmuted, respectively, there really isn't anything that requires actions from the user:

```diff
     api: Add route updateUserTopic

     For the legacy case, there can be an error when muting a topic that
-    is already muted or unmuting one that is already unmuted.  Let it
-    throw because we can't reliably filter out the error, which doesn't
-    have a specific "code".
+    is already muted or unmuting one that is already unmuted.
+
+    The callers are not expected to handle such errors because they aren't
+    really actionable.
```